nixos/modules/services/matrix/matrix-authentication-service.nix

```
@@ -0,0 +1,291 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    concatMapStringsSep
+    filterAttrsRecursive
+    getExe
+    mdDoc
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    optional
+    optionalAttrs
+    types
+    ;
+
+  cfg = config.services.matrix-authentication-service;
+  format = pkgs.formats.yaml { };
+
+  # remove null values from the final configuration
+  finalSettings = filterAttrsRecursive (_: v: v != null) cfg.settings;
+  configFile = format.generate "config.yaml" finalSettings;
+in
+{
+  options.services.matrix-authentication-service = {
+    enable = mkEnableOption (mdDoc "matrix authentication service");
+
+    package = mkPackageOption pkgs "matrix-authentication-service" { };
+
+    settings = mkOption {
+      default = { };
+      description = mdDoc ''
+        The primary mas configuration. See the
+        [configuration reference](https://element-hq.github.io/matrix-authentication-service/usage/configuration.html)
+        for possible values.
+
+        Secrets should be passed in by using the `extraConfigFiles` option.
+      '';
+      type = types.submodule {
+        freeformType = format.type;
+
+        options = {
+          http.public_base = mkOption {
+            type = types.str;
+            default = "http://[::]:8080/";
+          };
+          http.issuer = mkOption {
+            type = types.str;
+            default = "http://[::]:8080/";
+          };
+          http.trusted_proxies = mkOption {
+            type = types.listOf (types.str);
+            default = [ "192.168.0.0/16" "172.16.0.0/12" "10.0.0.0/10" "127.0.0.1/8" "fd00::/8" "::1/128" ];
```

```suggestion
            default = [ "127.0.0.1/8" "::1/128" ];
```

Just localhost by default, please. Also, how does this default play out when listening on a unix socket?

---

Took this default from [upstream docs](https://element-hq.github.io/matrix-authentication-service/setup/reverse-proxy.html?highlight=trusted#x-forwarded-for-header), will fix.

---

Fixed, regarding unix socket: I'll have to test to answer that question.